ambari-server/src/main/java/org/apache/ambari/server/state/SingleSignOnInfo.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ambari.server.state;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+
+/**
+ * {@link SingleSignOnInfo} encapsulates meta information about a service's support for single
+ * sign-on (SSO) integration
+ * <p>
+ * The data is expected to be like
+ * <pre>
+ *   &lt;sso&gt;
+ *     &lt;supported&gt;true&lt;/supported&gt;
+ *     &lt;enabledConfiguration&gt;config-type/sso.enabled.property_name&lt;/enabledConfiguration&gt;
+ *   &lt;/sso&gt;
+ * </pre>
+ */
+@XmlAccessorType(XmlAccessType.FIELD)
+public class SingleSignOnInfo {
+
+  /**
+   * Indicates whether the relevant service supports SSO integration (<code>true</code>) or not (<code>false</code>).
+   */
+  @XmlElement(name = "supported")
+  private Boolean supported = Boolean.FALSE;
+
+  /**
+   * The configuration that can be used to determine if SSO integration has been enabled.
+   * <p>
+   * It is expected that this value is in the form of <code>configuration-type/property_name</code>
+   */
+  @XmlElement(name = "enabledConfiguration")
+  private String enabledConfiguration = null;
```

Comment:
Is SSO support typically measurable by 1 property name? Also, no need to have a value either?

Reply:
I have only seen that relevant services have a "sso.enabled" Boolean property.  So I am starting there.  I am not sure how this will grow as more services support SSO integration, but I hope we can get them to add such a Boolean properly when they do. 

This can default to `null` since `null` is valid for `supported = false`.  There is a validator later on that checks to see if enabledConfiguration is empty when `supported = true`

Reply:
Yes, but eventually you'll hit the 1 service that is `sso.enabled=yes` or even worse `sso-type=None` / `sso-type=federated`. I don't think it needs to be this Jira since this code satisfies your needs, but it's something that will probably crop up in the future.

Reply:
I agree and will have to fight that fight when it comes. :(